### PR TITLE
3.0: Drop support for CentOS8

### PR DESCRIPTION
**Pull request by enrico-usai**
Note: we're preserving dnf and dnf-proxy logic in user data.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/resources/imagebuilder/parallelcluster.yaml [hunk 1]
```diff
@@ -280,10 +271,7 @@ phases:
               OS='{{ build.OperatingSystemName.outputs.stdout }}'
 
               if [[ ${!OS} =~ ^(centos[7-8]|ubuntu(18|20)04)$ ]]; then
```
Comment (hanwen-pcluste) [inline cli/src/pcluster/resources/imagebuilder/parallelcluster.yaml:273]: `if [[ ${!OS} =~ ^(centos7|ubuntu(18|20)04)$ ]]; then`

File: tests/integration-tests/tests/dcv/test_dcv.py [hunk 2]
```diff
@@ -33,11 +33,9 @@
 @pytest.mark.dimensions("us-gov-west-1", "c5.xlarge", "ubuntu1804", "slurm")
 @pytest.mark.dimensions("eu-west-1", "g3.8xlarge", "alinux2", "slurm")
 @pytest.mark.dimensions("eu-west-1", "g3.8xlarge", "centos7", "slurm")
-@pytest.mark.dimensions("eu-west-1", "g3.8xlarge", "centos8", "slurm")
```
Comment (hanwen-pcluste) [inline tests/integration-tests/tests/dcv/test_dcv.py:36]: Thank you for taking care of the integration tests. We will just need to remove all the annotations

File: cli/src/pcluster/resources/imagebuilder/parallelcluster.yaml (hunk 1 above)
Comment (enrico-usai) [reply cli/src/pcluster/resources/imagebuilder/parallelcluster.yaml]: Fixed

**Comment by enrico-usai** [comment]
Note: we're preserving dnf and dnf-proxy logic in user data

**Review by hanwen-pcluste — DISMISSED** [review]
LGTM!

**Review by hanwen-pcluste — APPROVED** [review]

**Pull request merged**